x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -852,7 +852,9 @@ private Expression replaceAliases(Expression condition, List<? extends NamedExpr
             return condition.transformDown(u -> {
                 boolean qualified = u.qualifier() != null;
                 for (Alias alias : aliases) {
-                    if (qualified ? Objects.equals(alias.qualifiedName(), u.qualifiedName()) : Objects.equals(alias.name(), u.name())) {
+                    if (qualified ? Objects.equals(alias.qualifiedName(), u.qualifiedName()) : Objects.equals(alias.name(), u.name())
```

Comment:
If I understand the issue correctly, the SO is caused by the unresolved references being replaced with aliases with the same name (which are duplicated) that are also unresolved which triggers a continuous loop.

I think the proper fix should not be here but in the method above that collects the `Alias`es for a given expression:
```
// a list might still be enough
           Set<Alias> aliases = new LinkedHashSet<>();
            named.forEach(n -> {
                if (n instanceof Alias) {
                    Alias a = (Alias) n;
                    if (a.resolved()) {
                         aliases.add(a);
                    }
                }
            })
```
This way unresolved references should be replaced only with resolved aliases that match their name (qualified or not).

Reply:
In this way `SELECT dete AS d FROM test WHERE d = 5` would generate an incorrect error message `Unknown column [d]` because [d] was never replaced as an alias and the original error (`Unknown column [dete]`) is lost.